src/main/cpp/CMakeLists.txt

```diff
@@ -144,6 +144,7 @@ add_library(
   src/CastStringJni.cpp
   src/NativeParquetJni.cpp
   src/cast_string.cu
+  src/cast_string_to_float.cu
```

Comment:
What is `cast_string.cu`? This new file name is more meaningful, thus we may want to rename the old file to make it more meaningful too.